src/main/java/com/zufar/onlinestore/payment/converter/PaymentMethodConverter.java

```diff
@@ -2,22 +2,27 @@
 
 import com.stripe.param.PaymentMethodCreateParams;
 import com.zufar.onlinestore.payment.dto.CreatePaymentMethodDto;
-import org.springframework.stereotype.Component;
+import org.mapstruct.Mapper;
 
-@Component
-public class PaymentMethodConverter {
+@Mapper
+public interface PaymentMethodConverter {
 
-    public PaymentMethodCreateParams toPaymentMethodParams(final CreatePaymentMethodDto createPaymentMethodDto) {
-        PaymentMethodCreateParams.CardDetails cardDetails = PaymentMethodCreateParams.CardDetails.builder()
-                .setNumber(createPaymentMethodDto.cardNumber())
-                .setExpMonth(createPaymentMethodDto.expMonth())
-                .setExpYear(createPaymentMethodDto.expYear())
-                .setCvc(createPaymentMethodDto.cvc())
-                .build();
+    default PaymentMethodCreateParams toPaymentMethodParams(final CreatePaymentMethodDto createPaymentMethodDto) {
+        if (createPaymentMethodDto == null) {
+            return null;
+        }
 
-        return PaymentMethodCreateParams.builder()
-                .setCard(cardDetails)
-                .setType(PaymentMethodCreateParams.Type.CARD)
-                .build();
+        PaymentMethodCreateParams.CardDetails.Builder cardDetails = PaymentMethodCreateParams.CardDetails.builder();
+        cardDetails.setNumber(createPaymentMethodDto.cardNumber());
+        cardDetails.setExpMonth(createPaymentMethodDto.expMonth());
+        cardDetails.setExpYear(createPaymentMethodDto.expYear());
+        cardDetails.setCvc(createPaymentMethodDto.cvc());
+
+        PaymentMethodCreateParams.Builder builder = PaymentMethodCreateParams.builder();
```

Comment:
Do we really need Builders for Mapstruct? 
I sow that Mapstruct provides set of annotations which we can use in our app to avoid boilerplate code